exec/java-exec/src/main/java/org/apache/drill/exec/ops/FragmentContext.java

```diff
@@ -245,6 +245,16 @@ public SchemaPlus getRootSchema() {
   }
 
   /**
+   * Returns the statement type (e.g. SELECT, CTAS, ANALYZE) from the query context.
+   * @return query statement type {@link QueryContext.StatementType}, if known.
+   */
+  public QueryContext.StatementType getStatementType() {
+    if (queryContext != null) {
```

Comment:
From existing code (getRootSchema()), it seems like if we have a non-root fragment the query context maybe null. I am not sure about the the conditions - hence the paranoia check. I can throw an exception just like getRootSchema().